dummydataServer/controller/userController.js

```diff
@@ -0,0 +1,208 @@
+        }
+        if (!userExist) {
+            res.status(404).json("user not found")
+      }
+        if (bookExist && bookExist.bookStatus === "unavailable") {
+            res.status(404).json({message: "This book is currently unavailable", "bookName": bookExist.bookName, 
+            "bookId": bookId, "status": bookExist.bookStatus, "username": userExist.username, "userId": userId})
+        }
+        else {
+             review.push(reviewerId, bookIden, req.body.reviewText)
+                res.json({ status: true, message:'Thanks for the review!', "bookName": bookExist.bookName, "bookId": bookIden, 
+                "username": userExist.userName, "userId": reviewerId, "reviewText":req.body.reviewText, "reviewId":reviewId} )
+        }
+    },
+    //API to mark a book as favorites
+    markAsFavorites (req, res)  {
+        let favoritesId = favorites.length + 1;
+        const reviewerId = parseInt(req.params.userId, 10);
+        const bookIden = parseInt(req.params.bookId, 10);
+        let userExist;
+	    let bookExist; 
+        user.forEach((u) =>{
+          const userNumber = parseInt(u.userId,10);
+          if (userNumber === reviewerId)
+           {
+            userExist= u;
+            return userExist
+           }
+        });
+        
+        books.forEach((book) =>{
+          const bookNumber = parseInt(book.bookId,10);
+          if (bookNumber === bookIden)
+          {
+            bookExist= book;
+            return bookExist
+          }
+        });
+        if (!bookExist) {
+            res.status(404).json("This book does not exist")
+        }
+        if (!userExist) {
+            res.status(404).json("user not found")
+      }
+      else{
+          favorites.push(bookIden, reviewerId)
+          res.status(200).json({status: true, message:"Book marked as favorites", "bookName": bookExist.bookName, "bookId": bookIden, 
+                "username": userExist.userName, "userId": reviewerId, "favoritesId": favoritesId
+          })
+      }
+    },
+    getFavoritesBook (req, res) {
+        const reviewerId = parseInt(req.params.userId, 10);
+        let userExist;
+        favorites.forEach((f) =>{
+          const userNumber = parseInt(f.userId,10);
+          if (userNumber === reviewerId)
+           {
+            userExist= f;
+            return userExist
+           }
+        });
+        if (!userExist) {
+            res.status(404).json("user not found")
+      }
+      else{
+        let userFavorites = [];
+        let favbooks = userExist.bookId
+        let userId = userExist.userId
+      userFavorites.push(favbooks)
+       res.status(200).json({message: "Your favorites book", "user": userId, "favbook": favbooks});
+    }
+      },
+      getUpvote (req, res){
+        const upvoteArr =[]
+        sortVotes = books.sort((a,b) => b.upvote - a.upvote)
+        upvoteArr.push(sortVotes)
+        res.status(200).json({message: true, "books": sortVotes})
+      }
+    
+    }
+
```

Comment:
Your indentation is poor Mary. Always ensure your methods and classes are well indented.
You should heed to `Eslint` flags.

Reply:
I will